.github/workflows/publish_images.yml

```
@@ -14,81 +14,17 @@ on:
-            echo "Using custom version: ${{ github.event.inputs.custom_version }}"
-            if [[ ! "${{ github.event.inputs.custom_version }}" =~ $VERSION_REGEX ]]; then
-              echo "Error: custom_version must match semantic versioning (e.g., v1.2.3)."
-              exit 1
-            fi
-            echo "::set-output name=release_version::${{ github.event.inputs.custom_version }}"
-          elif [[ "${GITHUB_REF}" == refs/tags/* ]]; then
-            echo "Using tag reference: ${GITHUB_REF#refs/tags/}"
-            echo "::set-output name=release_version::${GITHUB_REF#refs/tags/}"
-          else
-            echo "Defaulting to branch name: ${GITHUB_REF#refs/heads/}"
-            echo "::set-output name=release_version::${GITHUB_REF#refs/heads/}"
-          fi
-      - name: Get release version without prefix
-        id: get_release_version_without_prefix
-        env:
-          RELEASE_VERSION: ${{ steps.get_release_version.outputs.release_version }}
-        run: |
-          if [[ "${RELEASE_VERSION}" == v* ]]; then
-            echo "::set-output name=version_without_prefix::${RELEASE_VERSION:1}"
-          else
-            echo "::set-output name=version_without_prefix::${RELEASE_VERSION}"
-          fi
-      - name: Get highest semver
-        id: get_highest_semver
-        env:
-          RELEASE_VERSION: ${{ steps.get_release_version.outputs.release_version }}
-        run: |
-          if [[ -n "${{ github.event.inputs.custom_version }}" ]]; then
-            HIGHEST_SEMVER_TAG="${{ github.event.inputs.custom_version }}"
-            echo "::set-output name=highest_semver_tag::$HIGHEST_SEMVER_TAG"
-            echo "Using custom version as highest semantic version: $HIGHEST_SEMVER_TAG"
-          else 
-            source infra/scripts/setup-common-functions.sh
-            SEMVER_REGEX='^v[0-9]+\.[0-9]+\.[0-9]+(-([0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?$'
-            if echo "${RELEASE_VERSION}" | grep -P "$SEMVER_REGEX" &>/dev/null ; then
-              echo ::set-output name=highest_semver_tag::$(get_tag_release -m)
-              echo "Using infra/scripts/setup-common-functions.sh to generate highest semantic version: $HIGHEST_SEMVER_TAG"
-            fi
-          fi
-      - name: Check output
-        env:
-          RELEASE_VERSION: ${{ steps.get_release_version.outputs.release_version }}
-          VERSION_WITHOUT_PREFIX: ${{ steps.get_release_version_without_prefix.outputs.version_without_prefix }}
-          HIGHEST_SEMVER_TAG: ${{ steps.get_highest_semver.outputs.highest_semver_tag }}
-        run: |
-          echo $RELEASE_VERSION
-          echo $VERSION_WITHOUT_PREFIX
-          echo $HIGHEST_SEMVER_TAG
+    uses: ./.github/workflows/get_semantic_release_version.yaml
+    with:
+      custom_version: ${{ github.event.inputs.custom_version }}
+      token: ${{ github.event.inputs.token }}
 
   build-publish-docker-images:
     runs-on: ubuntu-latest
-    needs: [get-version]
+    needs: [ get-version ]
```

should this be added?
```suggestion
    needs: [ get-version, publish-python-sdk ]
```

---

I kept this out so that if we decided to run this component independently, it wouldn't trigger an SDK update.

Notice in the `publish.yml` file it does have these required (which is the end-to-end flow).